src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs

```diff
@@ -11,15 +11,23 @@ internal sealed class ProblemDetailsJsonOptionsSetup : IPostConfigureOptions<Jso
 {
     public void PostConfigure(string? name, JsonOptions options)
     {
-        if (options.SerializerOptions.TypeInfoResolver is not null)
+        switch (options.SerializerOptions.TypeInfoResolver)
         {
-            if (options.SerializerOptions.IsReadOnly)
-            {
-                options.SerializerOptions = new(options.SerializerOptions);
-            }
-
-            // Combine the current resolver with our internal problem details context
-            options.SerializerOptions.TypeInfoResolver = JsonTypeInfoResolver.Combine(options.SerializerOptions.TypeInfoResolver!, ProblemDetailsJsonContext.Default);
+            case DefaultJsonTypeInfoResolver:
+                // In this case, the current configuration is using a reflection-based resolver
+                // and we are prepending our internal problem details context to be evaluated
+                // first.
+                options.SerializerOptions.TypeInfoResolver = JsonTypeInfoResolver.Combine(ProblemDetailsJsonContext.Default, options.SerializerOptions.TypeInfoResolver);
+                break;
+            case not null:
+                // Combine the current resolver with our internal problem details context (adding last)
+                options.SerializerOptions.AddContext<ProblemDetailsJsonContext>();
```

Comment:
This is broken when someone calls `AddContext` before us right? So it'll be:
`custom context` -> `default context` -> `problem details context`